src/Storages/MergeTree/MergeTreeIndexText.cpp

```diff
@@ -173,17 +155,36 @@ PostingListPtr PostingsSerialization::deserialize(UInt64 header, UInt32 cardinal
         postings->addMany(cardinality, values.data());
         return postings;
     }
+    else
+    {
+        size_t num_bytes;
+        readVarUInt(num_bytes, istr);
 
-    size_t num_bytes;
-    readVarUInt(num_bytes, istr);
+        /// If the posting list is completely in the buffer, avoid copying.
+        if (istr.position() && istr.position() + num_bytes <= istr.buffer().end())
+            return std::make_shared<PostingList>(PostingList::read(istr.position()));
```

Comment:
Regarding my other comment that we have serialize methods for C and C++ Roaring, this creates the (slight) ugliness that we are reading using the C++ API (l. 165), but write using the C API (l. 111).